doc/design/overall.md

```diff
@@ -0,0 +1,20 @@
+# overall design
+
+Initial requirements:
+- Polling for GitHub deployment events
+- Download charts from a registry
+- Handling installation configuration from a configmap to charts
+- Install charts to the cluster
+
+Design decisions:
+- Use Helm charts for application definition, using a standard format is better than defining our own
+- Use Tiller for application templating and installation, this should save us replicating effort on templating and installation
+- Use GitHub deployment events for launching deployments, this is a decision that should fit in with our overall workflow
+- Open source by default. Don't force a specific configuration format, that is, organisations should be able to specify their own configuration schema
```

Comment:
We may want to have some of our operators in the official charts repo. There will probably also be 3rd party operators we want to run. So I think we should support both our own registry and the official one.